### ISSUE TITLE
DocumentCard: Title now receives keyboard focus, but doesn't have a visible focus indicator

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#22608]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
N/A
```

### Reproduction

docs site

### Bug Description

Related to #21461, now that the title node is the one to get keyboard focus, we should add a visible focus indicator, likely using `:focus-visible`.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
